CSC8503CoreClasses/JsonParser.cpp

```diff
@@ -0,0 +1,203 @@
+	}
+		break;
+
+	case RoomList:
+		level->mRoomList[Vector3(keyValuePairs[2]["x"], keyValuePairs[2]["y"], -keyValuePairs[2]["z"])] = Room((int)keyValuePairs[1]["type"]);
+		break;
+
+	case GuardCount:
+		level->mGuardCount = (int)keyValuePairs[0]["guardCount"];
+		break;
+
+	case CCTVCount:
+		level->mCCTVCount = (int)keyValuePairs[0]["cctvCount"];
+		break;
+
+	case GuardPaths:
+		level->mGuardPaths.push_back(std::vector<Vector3>());
+		for (int i = 2; i < keyValuePairs.size(); i++) {
+			level->mGuardPaths[level->mGuardPaths.size() - 1].push_back(Vector3(keyValuePairs[i]["x"], keyValuePairs[i]["y"], -keyValuePairs[i]["z"]));
+		}
+		break;
+
+	case CCTVTransforms:
+	{
+		Matrix4 newTransform = Matrix4::Rotation(keyValuePairs[3]["y"], Vector3(0, 1, 0)) *
+			Matrix4::Translation(Vector3(keyValuePairs[2]["x"], keyValuePairs[2]["y"], -keyValuePairs[2]["z"])) * Matrix4::Scale(Vector3(1, 1, 1)) * Matrix4();
+		if (level) level->mCCTVTransforms.push_back(newTransform);
+		else room->mCCTVTransforms.push_back(newTransform);
+	}
+		break;
+
+	case PrisonPosition:
+		level->mPrisonPosition = Vector3(keyValuePairs[1]["x"], keyValuePairs[1]["y"], -keyValuePairs[1]["z"]);
+		break;
+
+	case PlayerStartPositions:
+		for (int i = 0; i < MAX_PLAYERS; i++) {
+			if (!level->mPlayerStartPositions[i].x) {
+				level->mPlayerStartPositions[i] = Vector3(keyValuePairs[1]["x"], keyValuePairs[1]["y"], -keyValuePairs[1]["z"]);
+				break;
+			}
+		}
+		break;
+
+	case DirectionalLight:
+	{
+		Light* newLight = (Light*)new DirectionLight(Vector3(keyValuePairs[2]["x"], keyValuePairs[2]["y"], -keyValuePairs[2]["z"]),
+			Vector4(keyValuePairs[3]["x"], keyValuePairs[3]["y"], keyValuePairs[3]["z"], keyValuePairs[3]["w"]));
+		level->mLights.push_back(newLight);
+	}
+		break;
+
+	case Pointlight:
+	{
+		Light* newLight = (Light*)new PointLight(Vector3(keyValuePairs[2]["x"], keyValuePairs[2]["y"], -keyValuePairs[2]["z"]),
+			Vector4(keyValuePairs[3]["x"], keyValuePairs[3]["y"], keyValuePairs[3]["z"], keyValuePairs[3]["w"]), keyValuePairs[1]["radius"]);
+			if (level) level->mLights.push_back(newLight);
+			else room->mLights.push_back(newLight);
+	}
+		break;
+
+	case Spotlight:
+	{
+		Light* newLight = (Light*)new SpotLight(Vector3(keyValuePairs[4]["x"], keyValuePairs[4]["y"], -keyValuePairs[4]["z"]),
+			Vector3(keyValuePairs[2]["x"], keyValuePairs[2]["y"], -keyValuePairs[2]["z"]),
+			Vector4(keyValuePairs[3]["x"], keyValuePairs[3]["y"], keyValuePairs[3]["z"], keyValuePairs[3]["w"]),
+			keyValuePairs[1]["radius"], keyValuePairs[1]["angle"], 1.0f);
+		if (level) level->mLights.push_back(newLight);
+		else room->mLights.push_back(newLight);
+	}
+		break;
+
+	case ItemPositions:
+	{
+		Vector3 newPos = Vector3(keyValuePairs[1]["x"], keyValuePairs[1]["y"], -keyValuePairs[1]["z"]);
+		if (level) level->mItemPositions.push_back(newPos);
+		else room->mItemPositions.push_back(newPos);
+	}
+		break;
+
+	case Vents:
+		Vent* vent = new Vent();
```

Comment:
Same, `vent` have no delete or release.
If it is smart pointer or I am wrong, please ignore it.